### PR TITLE
task 9

### PR DESCRIPTION
**Pull request by annc4st**

### CODE REVIEW

File: controllers/controllers.js [hunk 1]
```diff
@@ -117,6 +117,23 @@ exports.patchArticle = (req, res, next) => {
     })
 };
 
+//9
+exports.deleteComment = (req, res, next) => {
+  const {comment_id} = req.params;
+  console.log("controllers commetnID ::", comment_id);
```
Comment (itsaliabdallah) [inline controllers/controllers.js:123]: remember to remove console.logs

File: controllers/controllers.js [hunk 2]
```diff
@@ -117,6 +117,23 @@ exports.patchArticle = (req, res, next) => {
...
+  if (isNaN(comment_id)) {
+    return res.status(400).send({ message: 'Invalid comment_id' });
+    
+}
```
Comment (itsaliabdallah) [inline controllers/controllers.js:127]: Does PSQL not throw an error for this?

File: models/models.js [hunk 3]
```diff
@@ -109,9 +108,19 @@ exports.updateArticle = async (article_id, inc_votes) => {
...
 };
 
- 
\ No newline at end of file
+ //9
+ exports.delComment = async (id) => {
+  const commentExists = await db.query(`SELECT * FROM comments WHERE comment_id = $1;`, [id])
+  if (commentExists.rows.length === 0) {
+    return Promise.reject({ status: 404, message: 'Comment does not exist' });
+  }
```
Comment (itsaliabdallah) [inline models/models.js:120]: Nice!

**Review by itsaliabdallah — APPROVED** [review]
Everything looks good.
Left some comments but feel free to merge when ready.

File: controllers/controllers.js (hunk 2 above)
Comment (annc4st) [reply controllers/controllers.js:127]: psql throws an error for this// i thought i need also to write it in controller
app.use((error, req, res, next) => {
  if (error.code === '22P02') {
    res.status(400).send({ message: 'Invalid input syntax'})
  }...

image

Comment (itsaliabdallah) [reply controllers/controllers.js]: If you have a PSQL error handler set up ready to recieve then we don't also need to send a new res.send

**Review by itsaliabdallah — APPROVED** [review]
All looking good.
Merge when ready.

**Pull request merged**